cpp/src/arrow/ipc/json-internal.cc

```diff
@@ -372,27 +372,42 @@ class ArrayWriter {
   template <typename T>
   typename std::enable_if<IsSignedInt<T>::value, void>::type WriteDataValues(
       const T& arr) {
+    static const char null_string[] = "0";
     const auto data = arr.raw_values();
-    for (int i = 0; i < arr.length(); ++i) {
-      writer_->Int64(data[i]);
+    for (int64_t i = 0; i < arr.length(); ++i) {
+      if (arr.IsValid(i)) {
+        writer_->Int64(data[i]);
+      } else {
+        writer_->RawNumber(null_string, sizeof(null_string));
```

Comment:
No tests should be asserting that the value in the null slots is coming back as 0, though, right?

Reply:
I don't think so, indeed. But to be frank I don't know anything about our JSON format. Here the goal is to avoid 1) reading uninitialized memory 2) outputting junk values in the JSON (even if they'll be ignored since the corresponding validity value is false).

Reply:
Got it, make sense. Thanks!